sources/osgGA/CADManipulatorStandardMouseKeyboardController.js

```diff
@@ -0,0 +1,186 @@
+'use strict';
+var OrbitManipulator = require( 'osgGA/OrbitManipulator' );
+
+var CADManipulatorStandardMouseKeyboardController = function ( manipulator ) {
+    this._manipulator = manipulator;
+    this._timer = false;
+    this.init();
+};
+
+CADManipulatorStandardMouseKeyboardController.prototype = {
+    init: function () {
+        this.releaseButton();
+        this._rotateKey = 65; // a
+        this._zoomKey = 83; // s
+        this._panKey = 68; // d
+        this._mode = undefined;
+    },
+    getMode: function () {
+        return this._mode;
+    },
+    setMode: function ( mode ) {
+        this._mode = mode;
+    },
+    setEventProxy: function ( proxy ) {
+        this._eventProxy = proxy;
+    },
+    setManipulator: function ( manipulator ) {
+        this._manipulator = manipulator;
+    },
+    setDimensionMask: function ( dimMask ) {
+        this._dimensionMask = dimMask;
+    },
+
+    mousemove: function ( ev ) {
+        if ( this._buttonup === true ) {
+            return;
+        }
+
+        var manipulator = this._manipulator;
+        var pos = manipulator.getPositionRelativeToCanvas( ev.clientX, ev.clientY );
+
+        if ( isNaN( pos[ 0 ] ) === false && isNaN( pos[ 1 ] ) === false ) {
+
+            var mode = this.getMode();
+            if ( mode === OrbitManipulator.Rotate ) {
+                manipulator.getRotateInterpolator().setTarget( pos[ 0 ], pos[ 1 ] );
+
+            } else if ( mode === OrbitManipulator.Pan ) {
+                manipulator.getPanInterpolator().setTarget( pos[ 0 ], pos[ 1 ] );
+
+            } else if ( mode === OrbitManipulator.Zoom ) {
+                var zoom = manipulator.getZoomInterpolator();
+                manipulator.computeIntersections( pos );
+
+                if ( zoom.isReset() ) {
+                    zoom._start = pos[ 1 ];
```

Comment:
Ok what I meant is that it's bad to access _member of extern class instance without accessors. So If you add 'private' member, it's better to add accessors to modify the member.
like setStart , getStart
But I know it's not your original code